### DISCUSSION TITLE
User Operator User with underscore in name

### DISCUSSION THREAD

**Discussion by yasin-amadmia-mck** [strimzi/strimzi-kafka-operator#5684] [Q&A] [answered]
Currently when we try to create a user with an underscore, the Topic Operator fails with error:
```
Error from server (Invalid): error when creating "aks-kafka-resources.yml": KafkaUser.kafka.strimzi.io "test_kafka_user" is invalid: metadata.name: Invalid value: "test_kafka_user": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```
This is coming from Kubernetes end I believe and the **Topic Operator** also has got same issue however, in **TopicOperator** it can be circumvented by using `topicName` within `specs`. (see https://github.com/strimzi/strimzi-kafka-operator/issues/2931)

Question: Is there any schema property like `userName` that we can use ? How to get a user with `_` work via the User Operator.

Thanks.

**Comment by scholzj** [comment, marked as answer]
You have to follow the Kubernetes rules for the user names. If you really need a users with incompatible names, you can also disable the User Operator and manage the users directly in Kafka yourself. In that case you are limited only by what Kafka defines as a valid username.
